### ISSUE TITLE
Istio 1.5.2 WASM filters cause Envoy Ingressgateway crash when EnvoyFilter with multiple ConfigPatches is active (possibly browser-specific)

### ISSUE THREAD

**Issue by JorenZ** [istio/istio#23609]
**TL;DR: our Istio 1.5.2 Envoy ingressgateway started crashing once we gave it an EnvoyFilter with multiple ConfigPatches (but only when visiting from Firefox, not from Chrome/Safari). Removing two Istio 1.5 WASM EnvoyFilters from the cluster solved this.**

**Bug description**
We encountered reproducible crashes in the ingressgateway, where the Envoy pod dies and restarts:
* initial Istio 1.5.2 update: no issues
* changing our Ingressgateway EnvoyFilter from one ConfigPatch to multiple ConfigPatches: gateway crashes when visited from my Ubuntu/Firefox 76.0 browser (but not when colleagues visited it with Apple/Safari, Apple/Chrome, neither when I visited it with Chrome from my Ubuntu machine)
* removing the new Istio 1.5 EnvoyFilters `metadata-exchange-1.5` and `stats-filter-1.5` in the `istio-system` namespace: no issues again, this seems to have solved it

I've attached the old EnvoyFilter on our gateway (everything worked), our new EnvoyFilter on our gateway (caused crashes initially, but this same filter is working now after removing the Istio WASM EnvoyFilters as above), and an example crash log from the pod.
 

**Expected behavior**
Istio works fine when you add an EnvoyFilter that applies multiple ConfigPatches. Specifically, the ingressgateway does not crash when visiting it with Firefox specifically (whereas Chrome/Safari seems to have always worked, see above).

**Steps to reproduce the bug**
* we run a management ingressgateway for Kiali and Kubernetes dashboard, and we upgraded our cluster to Istio 1.5.2 with seemingly no issues
* we used to run one EnvoyFilter for the whole gateway that allows an Authorization header and several request headers from our OAuth2 proxy (`envoy_filter_old.yaml`, attached)
* we added Grafana to the cluster. Grafana did not like the Authorization header so we split the EnvoyFilter to be SNI-specific: Kiali and Kubernetes want the Authorization header, Grafana does not. We created `envoy_fiilter_new.yaml` (attached), and then we saw the new crash behavior appear.
* we concluded by trial and error that deleting the two Istio-created WASM filters `metadata-exchange-1.5` and `stats-filter-1.5` solved our problem: `envoy_filter_new.yaml` is currently in effect and working as it should.
* this leaves our cluster with the following active EnvoyFilters, and seemingly no issues:
```
NAMESPACE      NAME
istio-system   metadata-exchange-1.4                    <-- left in: does nothing since it targets an old proxy version
istio-system   stats-filter-1.4                         <-- left in: does nothing since it targets an old proxy version
istio-system   tcp-metadata-exchange-1.5                <-- left in: does not seem to cause issues
istio-system   tcp-stats-filter-1.5                     <-- left in: does not seem to cause issues
kubeflow       kubeflow-ingressgateway-authentication   <-- contents as per `envoy_filter_old.yaml`: an ingressgateway for our Kubeflow deployment, not mentioned above but identical to `envoy_filter_old` except with a different selector/namespace
management     management-ingressgateway-authentication <-- contents as per `envoy_filter_new.yaml`
```

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**
```yaml
joren@laptop-joren:~$ istioctl version --remote
client version: 1.5.2
cluster-local-gateway version: 
cluster-local-gateway version: 
cluster-local-gateway version: 
cluster-local-gateway version: 
cluster-local-gateway version: 
pilot version: 1.5.2
data plane version: 1.3.1 (5 proxies), 1.5.2 (24 proxies)
```
(Note: we run Kubeflow on the cluster which has spawned 5 instances of the `cluster-local-gateway`, which is STALE since it's running a very old proxy version. This does not seem relevant for the bug at hand; it's been like this before the bug occurred and it's like this now that the bug has gone away again.)

```
joren@laptop-joren:~$ kubectl version
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.0", GitCommit:"70132b0f130acc0bed193d9ba59dd186f0e634cf", GitTreeState:"clean", BuildDate:"2019-12-07T21:20:10Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.9", GitCommit:"2e808b7cb054ee242b68e62455323aa783991f03", GitTreeState:"clean", BuildDate:"2020-01-18T23:24:23Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
```

**How was Istio installed?**
Via Ansible: we run `istioctl manifest apply -f our_manifest.yaml` with a custom IstioOperator yaml

**Environment where bug was observed (cloud vendor, OS, etc)**
On-premise cloud with Ubuntu 18.04 machines

[pod_crash_log.txt](https://github.com/istio/istio/files/4591697/pod_crash_log.txt)
[envoy_filter_old.yaml.txt](https://github.com/istio/istio/files/4591699/envoy_filter_old.yaml.txt)
[envoy_filter_new.yaml.txt](https://github.com/istio/istio/files/4591700/envoy_filter_new.yaml.txt)
